crates/model/src/order.rs

```diff
@@ -350,6 +350,11 @@ impl OrderData {
             self.valid_to,
         )
     }
+
+    /// Checks if the order is a market order.
+    pub fn within_market(&self, quote_sell_amount: &U256, quote_buy_amount: &U256) -> bool {
```

Comment:
nit: you can extend OrderData inside the `order_quoting.rs` module to allow using the quote type as an argument here which makes this nicer to read.

```rs
pub trait OrderPricing {
    fn within_market(&self, quote: &Quote) -> bool;
}
impl OrderPricing for OrderData {
    /// Checks if the order is a market order.
    fn within_market(&self, quote: &Quote) -> bool {
        self.sell_amount.full_mul(quote.buy_amount) >= quote.sell_amount.full_mul(self.buy_amount)
    }
}
```